src/org/omegat/core/search/Searcher.java

```diff
@@ -628,7 +631,24 @@ public boolean searchString(String origText, boolean collapseResults) {
             }
             while (true) {
                 int start = matcher.start();
-                foundMatches.add(new SearchMatch(start, matcher.end()));
+                if (m_searchExpression.mode == SearchMode.REPLACE) {
+                    if (m_searchExpression.searchExpressionType == SearchExpression.SearchExpressionType.REGEXP) {
+                        String repl = m_searchExpression.replacement;
+                        Matcher replaceMatcher = PatternConsts.REGEX_VARIABLE.matcher(repl);
+                        while (replaceMatcher.find()) {
+                            int varId = Integer.parseInt(replaceMatcher.group(1));
+                            repl = repl.substring(0, replaceMatcher.start())
+                                + matcher.group(varId) // yes, from source matcher!
+                                + repl.substring(replaceMatcher.end());
+                            replaceMatcher.reset(repl);
+                        }
+                        foundMatches.add(new SearchMatch(start, matcher.end(), StringUtil.replaceCase(repl, Core.getProject().getProjectProperties().getTargetLanguage().getLocale())));						
```

Comment:
Please humor my nitpicking here:

In general in the OmegaT codebase there is too much referencing of `Core`, especially for project properties. It's basically a big ball of global variables, and it overly couples things to the OmegaT concept of a project. One far-off goal of mine is to decouple all of this so that we can modularize OmegaT and e.g. offer just an API dependency for plugins, etc.

In the case of the `Searcher`, there's no reason a `Searcher` should have to know or care about what a project is. So instead of accessing the locale through `Core`, I think the `Searcher` should receive the locale at the point of instantiation via its constructor and keep it in a private field.

Granted, `Searcher` already looks to `Core` for a few other things, but I would like to keep those things from growing.

Reply:
In the meantime I found that the project is already in the private fields. So, we can replace Core.getProject() by m_project.

Reply:
Ah yes, that's perfect. I missed that. Thanks.